### ISSUE TITLE
CpGrid does not adhere to the DUNE grid interface 

### ISSUE THREAD

**Issue by andlaus** [OPM/dune-cornerpoint#150]
At least it does not adhere to the interface of Dune 2.4 because the `gridcheck()` function which is shipped with Dune does not even compile:

```c++
#include "config.h"

#include <dune/grid/CpGrid.hpp>

#include <opm/parser/eclipse/Parser/Parser.hpp>
#include <opm/parser/eclipse/Deck/Deck.hpp>

#define DISABLE_DEPRECATED_METHOD_CHECK 1
#if DUNE_VERSION_NEWER(DUNE_GRID, 2,4)
#include <dune/grid/test/gridcheck.hh>
#else
#include <dune/grid/test/gridcheck.cc>
#endif

#include <dune/common/parallel/mpihelper.hh>

const char *deckString =
    "RUNSPEC\n"
    "METRIC\n"
    "DIMENS\n"
    "2 2 2 /\n"
    "GRID\n"
    "DXV\n"
    "2*1 /\n"
    "DYV\n"
    "2*1 /\n"
    "DZ\n"
    "8*1 /\n"
    "TOPS\n"
     "4*100.0 /\n";

int main(int argc, char **argv)
{
    Dune::MPIHelper::instance(argc, argv);

    Opm::Parser parser;
    const auto deck = parser.parseString(deckString);

    Dune::CpGrid grid;
    grid.processEclipseFormat(deck, false);

    gridcheck(grid);
}
```

currently this makes the compiler produce something like this:

```
In file included from /home/and/src/dune-cornerpoint/tests/cpgrid/mytest.cpp:10:
In file included from /home/and/src/dune-grid/dune/grid/test/gridcheck.hh:20:
/home/and/src/dune-grid/dune/grid/test/staticcheck.hh:59:28: error: no type named
      'EntitySeed' in 'Dune::cpgrid::Entity<0>'
  typedef typename Entity::EntitySeed EntitySeed DUNE_UNUSED;
[...]
```

amongst quite a few other errors.

**Comment by dr-robertk** [comment]
I can take a look at this.

**Comment by dr-robertk** [comment]
This should be fixed in #183.
